light-client/src/tests.rs

```diff
@@ -75,6 +82,72 @@ impl Trusted {
     }
 }
 
+#[derive(Clone)]
+pub struct MockClock {
+    pub now: Time,
+}
+
+impl Clock for MockClock {
+    fn now(&self) -> Time {
+        self.now
+    }
+}
+
+#[derive(Clone)]
+pub struct MockIo {
+    chain_id: String,
+    light_blocks: HashMap<Height, LightBlock>,
+    latest_height: Height,
+}
+
+impl MockIo {
+    pub fn new(chain_id: String, light_blocks: Vec<LightBlock>) -> Self {
+        let latest_height = light_blocks.iter().map(|lb| lb.height()).max().unwrap();
+
+        let light_blocks = light_blocks
+            .into_iter()
+            .map(|lb| (lb.height(), lb))
+            .collect();
+
+        Self {
+            chain_id,
+            light_blocks,
+            latest_height,
+        }
+    }
+}
+
+#[contract_trait]
+impl Io for MockIo {
+    fn fetch_light_block(&self, _peer: PeerId, height: AtHeight) -> Result<LightBlock, IoError> {
+        let height = match height {
+            AtHeight::Highest => self.latest_height,
+            AtHeight::At(height) => height,
+        };
+
+        self.light_blocks
+            .get(&height)
+            .cloned()
+            .ok_or_else(|| rpc::Error::new((-32600).into(), None).into())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct MockEvidenceReporter;
+
+#[contract_trait]
+impl EvidenceReporter for MockEvidenceReporter {
+    fn report(&self, _e: Evidence, _peer: PeerId) -> Result<Hash, IoError> {
+        Ok(Hash::new([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]))
```

Comment:
Rust has a [short-hand](https://doc.rust-lang.org/stable/rust-by-example/primitives/array.html) for initializing arrays and slices with the same value :)

```suggestion
        Ok(Hash::new([0; 32]))
```

Reply:
🤦‍♀️